### PR TITLE
Exercise 3

### PR DESCRIPTION
**Pull request by MaycieS**

### CODE REVIEW

**Review by larskotthoff — CHANGES_REQUESTED** [review]
Sorry, don't really understand what you're doing at all.

For example, you say "After performing a prediction line, the mean
error was 227.6 degrees with an accuracy of 94.41%."

What's a prediction "line"? How do you measure error in degrees? You say you're predicting the body mass, so how can you measure accuracy? What are the "scores"?

What hyperparameters did you tune? How did you split your data?

**Review by larskotthoff — CHANGES_REQUESTED** [review]
Sorry, still don't understand how you can measure error in degrees. Degrees of what? This is not a common ML metric.

And please replace the reports instead of adding new ones.

**Review by larskotthoff — CHANGES_REQUESTED** [review]
Still don't understand what you're doing. Are you predicting the flipper length? The max value for that is 231, so errors of almost 4000 and 227.6 seem extremely large. Is this the squared error?

Please state at the beginning very clearly the ML problem you're solving -- what are you predicting based on what and how you're measuring error, i.e. what metric you are using.

**Review by larskotthoff — CHANGES_REQUESTED** [review]
Sorry, still don't know what you're doing.

You say you're predicting the body mass, but you're also giving an accuracy of 34.33%?

And when you're converting the categorical features to numbers, you're introducing artificial relationships -- 2 is twice as large as 1, but the original categories don't have those. I would recommend to use a one-hot-encoding or something similar.

**Review by larskotthoff — CHANGES_REQUESTED** [review]
What statistical tests did you run to assess overfitting and what was the result of those tests?

**Review by larskotthoff — APPROVED** [review]
